File: spoj/cubeRoot.py
N = input()
while 1:
	try:
		check = 0;
		i = 0
		count = 0
		string2 = ""
		x = int(raw_input())
		N -= 1
		string =  '%.12f' % pow(x, 0.3333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333)
		for c in string:
			string2 += c
			if c == '.':
				count = 1
			elif ord(c) >= ord('0') and ord(c) <= ord('9'):
				check = check + (ord(c) - 48)
				if count:i += 1
			if i == 10: break
		print check % 10, string2
	except:continue
	if not N:break

File: spoj/cubeRoot.cpp
/*******************************************************************************
 * Author       : Kashyap R Puranik
 * email        : kashyap.r.puranik (at) gmail (dot) com
 * copyright    : 2008 - 2009
 * date         : 23 - 08 -09
 * 
 * file name    : .cpp 
 * version      : 1.0.1
 ******************************************************************************/
 
#include <cstdio>
#include <cstdlib>
#include <cmath>

int main()
{
	int t, T;										// Test cases and test case counter
	scanf("%d",&T);									// Get the number of test cases
	
	for ( t = 0 ; t < T ; t ++ )					// Go through all test cases
	{
		double in, root;							// Input and output
		char str[100];								// String form of the cube root
		int i = 0;									// Loop counter for char array index
		int check = 0;
		/// Input
		while(! scanf("%lf",&in));					// Scan all input
		/// Calculation
		root = pow(in,0.333333333333333333333);		// Calculate cube root
		sprintf(str,"%.12lf\n",root);
		
		while(1)
		{
			if(str[i] == '.')
				break;
			check += str[i] - '0';
		}
		/// Output
		
	}
	
	return 0;										// Successful termination
}

